mesa/agent.py

```diff
@@ -115,18 +108,11 @@ def __init__(self, agents: Iterable[Agent], model: Model):
             agents (Iterable[Agent]): An iterable of Agent objects to be included in the set.
             model (Model): The ABM model instance to which this AgentSet belongs.
         """
-        self.model = model
+        self._agents = weakref.WeakKeyDictionary()
+        for agent in agents:
+            self._agents[agent] = None
```

Comment:
why not do this loop as a dict expression as previously?

Reply:
Not intended. Got a merge conflict, solved it the wrong way. Should be fixed now.